src/messages/mod.rs

```diff
@@ -306,26 +306,9 @@ impl SignedRoutingMessage {
 
         match self.content.src {
             ClientManager(_) | NaeManager(_) | NodeManager(_) => {
-                // Note: there should be exactly one source section, but we use safe code:
-                let valid_names: HashSet<_> = self
-                    .src_section
-                    .iter()
-                    .flat_map(|si| si.members().iter().map(PublicId::name))
-                    .sorted_by(|lhs, rhs| self.content.src.name().cmp_distance(lhs, rhs))
-                    .into_iter()
-                    .take(min_section_size)
-                    .collect();
-                let valid_sigs = self
-                    .signatures
-                    .sigs
-                    .keys()
-                    .filter(|pub_id| valid_names.contains(pub_id.name()))
-                    .count();
-                // TODO: we should consider replacing valid_names.len() with
-                // cmp::min(routing_table.len(), min_section_size)
-                // (or just min_section_size, but in that case we will not be able to handle user
-                // messages during boot-up).
-                valid_sigs * QUORUM_DENOMINATOR > valid_names.len() * QUORUM_NUMERATOR
+                let valid_sigs = self.signatures.len();
+                valid_sigs * QUORUM_DENOMINATOR > self.src_size() * QUORUM_NUMERATOR
+                    && self.src_size() >= min_section_size
```

Comment:
Not sure about that `&& self.src_size() >= min_section_size` why do they not share the exact same condtion as `Section(_)`

Reply:
My understanding is that the `Section` could is growing, hence the number could be `lower` than the min_sec_size.
Meanwhile, an `Authority` is expected to be `at least more than min_size_size`

Reply:
I think this `at least more than min_size_size` was only relevant before when we were selecting a close group. in the case you highlight. Now that there is no sub group, the restriction should be the same as for Section Authority because they are the same.